tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -926,4 +939,86 @@ public function renameGroup(string $oldGroup, string $newGroup, ?string $user =
 	public function userRenamesGroupUsingTheGraphApi(string $user, string $oldGroup, string $newGroup): void {
 		$this->featureContext->setResponse($this->renameGroup($oldGroup, $newGroup, $user));
 	}
+
+	/**
+	 * @param string $user
+	 *
+	 * @return ResponseInterface
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function retrieveUserInformationUsingGraphApi(
+		string $user
+	):ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		return GraphHelper::getOwnInformationWithGroups(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials["username"],
+			$credentials["password"],
+		);
+	}
+
+	/**
+	 * @When /^the user "([^"]*)" retrieves (her|his) information using the Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws JsonException
+	 */
+	public function userRetrievesHisorHerInformationOfUserUsingGraphApi(
+		string $user
+	):void {
+		$response = $this->retrieveUserInformationUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+    /**
+     * @Then /^the user retrieved API response should contain the following information:$/
+     *
+     * @param TableNode $table
+     *
+     * @return void
+     * @throws GuzzleException
+     */
+	public function theUserRetrieveApiResponseShouldContainTheFollowingInformation(TableNode $table): void
+    {
+        $rows = $table->getHash();
+        $apiResponse = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse());
+        foreach ($rows as $row) {
+            $this->checkUserInfromation($row, $apiResponse);
+        }
+    }
+
+    /**
+     * @param array $expectedValue
+     * @param array $actualValue
+     *
+     * @throws GuzzleException
+     * @return void
+     */
+    public function checkUserInfromation(array $expectedValue, array  $actualValue):void {
+        foreach (array_keys($expectedValue) as $keyName) {
+            if ($keyName === 'memberOf') {
```

Comment:
switch case will be more beneficial. https://stackoverflow.com/a/427835/11656407